MapboxCoreNavigation/RouteController.swift

```diff
@@ -296,21 +296,34 @@ open class RouteController: NSObject {
      */
     public var location: CLLocation? {
         guard let location = rawLocation else { return nil }
-        guard let stepCoordinates = routeProgress.currentLegProgress.currentStep.coordinates else { return nil }
-        guard let snappedCoordinate = Polyline(stepCoordinates).closestCoordinate(to: location.coordinate) else { return location }
 
         var nearByCoordinates = routeProgress.currentLegProgress.nearbyCoordinates
-        let nearByPolyline = Polyline(nearByCoordinates)
 
         // If the upcoming maneuver a sharp turn, only look at the current step for snapping.
         // Otherwise, we may get false positives from nearby step coordinates
         if let upcomingStep = routeProgress.currentLegProgress.upComingStep,
             let initialHeading = upcomingStep.initialHeading,
             let finalHeading = upcomingStep.finalHeading,
-            initialHeading.differenceBetween(finalHeading) < RouteControllerMaxManipulatedCourseAngle,
             let coordinates = routeProgress.currentLegProgress.currentStep.coordinates {
-           nearByCoordinates = coordinates
+            
+            // Calculate if angle is sharp
+            let inAngle = initialHeading * .pi / 180.0
+            let outAngle = finalHeading * .pi / 180.0
+            
+            let inX = sin(inAngle)
+            let inY = cos(inAngle)
+            let outX = sin(outAngle)
+            let outY = cos(outAngle)
+            
+            let turnAngle = acos((inX * outX + inY * outY) / 1.0) * (180 / .pi)
```

Comment:
Turf provides a `differenceBetween(_:)` method for calculating the difference between two angles.

Reply:
@1ec5 thats the issue here. The difference between a slight turn and a sharp turn can be the same angle. This is being removed in https://github.com/mapbox/mapbox-navigation-ios/pull/754/files#diff-1a2149692b3d7c58b6a0392120d6011aL310

Reply:
Ah, so the bug was that `differenceBetween(_:)` would take the shorter path around the circle, which we didn’t want in this case. I think we should upstream this version of the comparison to Turf.